atc/api/buildserver/abort.go

```diff
@@ -14,14 +13,7 @@ func (s *Server) AbortBuild(build db.Build) http.Handler {
 			"build": build.ID(),
 		})
 
-		engineBuild, err := s.engine.LookupBuild(aLog, build)
-		if err != nil {
-			aLog.Error("failed-to-lookup-build", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-
-		err = engineBuild.Abort(aLog)
+		err := build.MarkAsAborted()
```

Comment:
Related: https://github.com/concourse/concourse/issues/1585

Should I prioritize that soon while this is fresh in our heads?

Reply:
Maybe this could be something to look into when we do the broader LISTEN/NOTIFY work?